components/CompanyRegister.tsx

```diff
@@ -1,128 +1,145 @@
 import { CreateCompanyItem } from '@/db/entities/company';
 import useCompany from '@/hooks/useCompanyRegistration';
 import { isSuccessResponse } from '@/pages/api/companies';
+import { zodResolver } from '@hookform/resolvers/zod';
 import { useRouter } from 'next/router';
 import { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
 import { toast } from 'react-toastify';
 import { useAccount } from 'wagmi';
+import { z } from 'zod';
 import { useRecruitmentRegisterCompany } from '../src/generated';
 import { Badge } from './ui/badge';
 
+const schema = z.object({
+  companyName: z
+    .string()
+    .min(1, { message: 'Company Name is required' })
+    .max(255),
+  companyEmail: z.string().email({ message: 'Invalid email format' }),
+});
+
+interface FormData {
+  companyName: string;
+  companyEmail: string;
+  companySite: string;
+}
+
 const CompanyRegister = () => {
   const router = useRouter();
   const { address } = useAccount();
-  const { isValidating, isRegistered } = useCompany(address);
+  const { isRegistered } = useCompany(address);
+  const [isValidating] = useState(false);
```

Comment:
opps - isValidating comes from the useCompany hook :)

Reply:
Re-adjusted